sdk/ml/azure-ai-ml/azure/ai/ml/entities/_indexes/data_index_func.py

```
@@ -0,0 +1,743 @@
+    )
+    return component
+
+
+def get_component_obj(ml_client, component_uri):
+    from azure.ai.ml import MLClient
+
+    if not isinstance(component_uri, str):
+        # Assume Component object
+        return component_uri
+
+    matches = re.match(
+        r"azureml://registries/(?P<registry_name>.*)/components/(?P<component_name>.*)"
+        r"/(?P<identifier_type>.*)/(?P<identifier_name>.*)",
+        component_uri,
+    )
+    if matches is None:
+        from azure.ai.ml import load_component
+
+        # Assume local path to component
+        return load_component(source=component_uri)
+
+    registry_name = matches.group("registry_name")
+    registry_client = MLClient(
+        subscription_id=ml_client.subscription_id,
+        resource_group_name=ml_client.resource_group_name,
+        credential=ml_client._credential,
+        registry_name=registry_name,
+    )
+    component_obj = registry_client.components.get(
+        matches.group("component_name"),
+        **{matches.group("identifier_type").rstrip("s"): matches.group("identifier_name")},
+    )
+    return component_obj
+
+
+def _resolve_connection_id(ml_client, connection: Optional[Union[str, Connection]] = None) -> Optional[str]:
+    if connection is None:
+        return None
+
+    if isinstance(connection, str):
+        short_form = re.match(r"azureml:(?P<connection_name>[^/]*)", connection)
+        if short_form:
+            connection_name = short_form.group("connection_name")
+        else:
+            # TODO: Handle long form connection sub/rg/ws, ideally reuse logic implemented by connections code.
+            long_form = re.match(r"(azureml:/)?/.*/connections/(?P<connection_name>[^/]*)", connection)
+            connection_name = long_form.group("connection_name") if long_form else connection
+
+        try:
+            connection = ml_client.connections.get(connection_name)
+        except Exception:  # pylint: disable=broad-except
```

Do we actually need to swallow all exceptions here? Is there not a more specific one we're looking to circumvent?

---

In fact I don't think the `except Exception` clause is needed now. Thanks for catching it.